pkg/skaffold/kubernetes/manifest/image_pull_policy.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import apimachinery "k8s.io/apimachinery/pkg/runtime/schema"
+
+type ResourceSelectorImagePullPolicy struct{}
+
+func NewResourceSelectorImagePullPolicy() *ResourceSelectorImagePullPolicy {
+	return &ResourceSelectorImagePullPolicy{}
+}
+
+func (rs *ResourceSelectorImagePullPolicy) allowByGroupKind(gk apimachinery.GroupKind) bool {
```

Comment:
A short comment explaining the goal of the method and the criteria used would be helpful.